### DISCUSSION TITLE
Dyanmic Query Keys with React Query

### DISCUSSION THREAD

**Discussion by thusithagh** [TanStack/query#6366] [General]
I have this situation where I have to generate query keys to be used by react query.

`const datesToRequest = ['2023-11-01', '2023-11-02', '2023-11-03']`

Above is actually dynamically generated. For the ease of explaining I've hard coded it.

```
const vehiclesQuery = (dates: string[]) => useQuery({ 
   queryKey: ['dates', ...datesToRequest], 
   queryFn: () => fetchVehiclesByDate(dates) })

const vehicles = vehiclesQuery(datesToRequest)
```

`fetchVehiclesByDate` will call to the server and return some data. However, the `fetchVehiclesByDate` function is not seems to be getting called.
It works only when I have the query key is hard coded like below:

```
useQuery({ 
    queryKey: ['2023-11-01'], 
    queryFn: () => fetchVehiclesByDate(['2023-11-01']) })
```

Any idea on how to get this working for my case?

**Comment by TkDodo** [comment]
show a reproduction please. not sure if this is just a typo but:

```
   queryKey: ['dates', ...datesToRequest], 
   queryFn: () => fetchVehiclesByDate(dates) })
```

you're using `datesToRequest` in the key and `dates` in the fn
